traitsui/qt4/directory_editor.py

```diff
@@ -35,11 +35,11 @@ class SimpleEditor(SimpleFileEditor):
 
     def _create_file_dialog(self):
         """Creates the correct type of file dialog."""
-        dlg = QtGui.QFileDialog(self.control)
-        dlg.selectFile(self._file_name.text())
-        dlg.setFileMode(QtGui.QFileDialog.FileMode.Directory)
```

Comment:
pyface also seems to do `dlg.setViewMode(QtGui.QFileDialog.ViewMode.Detail)` but does not seem to set `dlg.setOptions(QtGui.QFileDialog.Option.ShowDirsOnly)`